Fix LICENSE and NOTICE for the kafka-connect-runtime distributions

jbonofre opened the pull request:
- Use specific `LICENSE`/`NOTICE` per kafka-connect-runtime distributions (main and hive)
- Update version
- Cleanup jars not found in the `lib` folder of the distribution

bryanck commented:
There are 2 different KC runtimes, one that includes an HMS client and one that doesn't, so with the current license/notice I took a conservative approach and included Hive, so not sure if we want to exclude Hive or not.

jbonofre commented:
@bryanck yes. I saw that. Both distributions use the same LICENSE/NOTICE so it won't be good either way 😆
Let me update with the largest common denominator. I'm checking.

jbonofre commented:
> There are 2 different KC runtimes, one that includes an HMS client and one that doesn't, so with the current license/notice I took a conservative approach and included Hive, so not sure if we want to exclude Hive or not.

I updated the PR to have specific `LICENSE`/`NOTICE` per distributions (main and hive). I also did a complete pass and updated versions/content.

amogh-jahagirdar commented on kafka-connect/kafka-connect-runtime/hive/NOTICE:
```diff
@@ -1517,7 +1317,7 @@ The Apache Software Foundation (http://www.apache.org/).
 
 --------------------------------------------------------------------------------
 
-Group: org.apache.kerby  Name: kerby-asn1  Version: 1.0.1
```
Same as above, since this is an ASF project do we even need to mention it in NOTICE?

jbonofre replied:
I added a commit with ASF projects mentions from `NOTICE`.

amogh-jahagirdar commented on kafka-connect/kafka-connect-runtime/hive/NOTICE:
```diff
@@ -1399,7 +1221,7 @@ The Apache Software Foundation (http://www.apache.org/).
 --------------------------------------------------------------------------------
 
-Group: org.apache.orc  Name: orc-shims  Version: 1.9.4
+Group: org.apache.orc  Name: orc-shims  Version: 1.9.5
```
Same as the Apache Kerby comment, since it's ASF we could technically just remove it from the notice? Non-blocking though since my understanding is it's not a must to not have it.

amogh-jahagirdar commented on kafka-connect/kafka-connect-runtime/hive/LICENSE:
```diff
@@ -219,113 +219,105 @@ License (from POM): The Apache Software License, Version 2.0 - http://www.apache
 
 --------------------------------------------------------------------------------
 
-Group: com.fasterxml.jackson.dataformat  Name: jackson-dataformat-xml  Version: 2.17.2
```
Confirmed we can remove this

amogh-jahagirdar commented on kafka-connect/kafka-connect-runtime/hive/LICENSE:
```diff
@@ -618,33 +610,21 @@ License (from POM): The Apache Software License, Version 2.0 - https://www.apach
 
 --------------------------------------------------------------------------------
 
-Group: com.squareup.okhttp3  Name: okhttp  Version: 4.9.3
-Project URL (from POM): https://square.github.io/okhttp/
-License (from POM): The Apache Software License, Version 2.0 - http://www.apache.org/licenses/LICENSE-2.0.txt
-
---------------------------------------------------------------------------------
-
-Group: com.squareup.okio  Name: okio  Version: 2.8.0
-Project URL (from POM): https://github.com/square/okio/
-License (from POM): The Apache Software License, Version 2.0 - http://www.apache.org/licenses/LICENSE-2.0.txt
-
---------------------------------------------------------------------------------
-
```
Confirmed we can remove this

amogh-jahagirdar commented on kafka-connect/kafka-connect-runtime/hive/LICENSE:
```diff
@@ -755,243 +735,303 @@ License (from POM): Apache License 2.0 - http://www.apache.org/licenses/LICENSE-
 
 --------------------------------------------------------------------------------
 
-Group: io.opencensus  Name: opencensus-api  Version: 0.31.1
-Project URL (from POM): https://github.com/census-instrumentation/opencensus-java
-License (from POM): The Apache License, Version 2.0 - http://www.apache.org/licenses/LICENSE-2.0.txt
```
I see this in the distribution

amogh-jahagirdar replied:
Ah okay, we added it back below, all good

amogh-jahagirdar reviewed (approved):
Thanks @jbonofre left some comments, but none of them are blocking. Mostly just removing the ASF projects from NOTICE since if I understand right we don't need them there. I went through both hive/main distributions and these changes align with what I'm seeing. It'd be good to get a second set of eyes on this change though.

Big picture question I have though is if it makes sense for Iceberg to even be releasing a separate Kafka Connect distribution with HMS client. Since we're close to a release, and I also just don't have context on the rationale I think the change to have separate license/notice for each distribution makes sense cc @bryanck . It's just that if later on as a community we decide that such a distribution is no longer is needed to be sourced from Iceberg we can take that opportunity to simplify some of the License/Notice aspects here.

jbonofre commented on kafka-connect/kafka-connect-runtime/hive/LICENSE:
```diff
@@ -1079,27 +1103,14 @@ Project URL (from POM): http://java.sun.com/products/jta
 
 --------------------------------------------------------------------------------
 
-Group: javax.xml.bind  Name: jaxb-api  Version: 2.2.2
-Project URL (from POM): https://jaxb.dev.java.net/
-License (from POM): CDDL 1.1 - https://glassfish.dev.java.net/public/CDDL+GPL_1_1.html
-License (from POM): GPL2 w/ CPE - https://glassfish.dev.java.net/public/CDDL+GPL_1_1.html
-
---------------------------------------------------------------------------------
```
Yes, we have both versions, but only 2.2.12 is actually in the distribution.

jbonofre commented on kafka-connect/kafka-connect-runtime/hive/NOTICE:
```diff
@@ -120,18 +120,7 @@ The Apache Software Foundation (http://www.apache.org/).
---------------------------------------------------------------------------------
-
-Group: org.apache.kerby  Name: kerb-crypto  Version: 1.0.1
+Group: org.apache.kerby  Name: kerb-crypto  Version: 2.0.3
```
I didn't cleanup ASF projects from `NOTICE` yet (wanted to have a first shot).

jbonofre commented:
I added:
1. Cleanup of ASF projects mentions from `NOTICE`
2. Add formatting in `NOTICE`

Fokko commented on kafka-connect/kafka-connect-runtime/main/LICENSE:
```diff
@@ -0,0 +1,1442 @@
+
+--------------------------------------------------------------------------------
+
+Group: com.github.pjfanning  Name: jersey-json  Version: 1.22
```
```suggestion
Group: com.github.pjfanning  Name: jersey-json  Version: 1.22.0
```

Fokko commented on kafka-connect/kafka-connect-runtime/main/LICENSE:
```diff
@@ -0,0 +1,1442 @@
+
+--------------------------------------------------------------------------------
+
+Group: javax.servlet.jsp  Name: jsp-api  Version: 2.1
```
Should we still point to the old repository? https://github.com/javaee/javaee-jsp-api

jbonofre replied:
Done

Fokko commented on kafka-connect/kafka-connect-runtime/main/LICENSE:
```diff
@@ -0,0 +1,1442 @@
+Project URL (from POM): https://aws.amazon.com/sdkforjava
+License (from POM): Apache License, Version 2.0 - https://aws.amazon.com/apache2.0
+
+--------------------------------------------------------------------------------
```
Looks like `http-auth-aws-crt` is missing:

```
./iceberg-kafka-connect-runtime-1.8.0-SNAPSHOT/lib/http-auth-aws-crt-2.30.11.jar
```

jbonofre replied:
Done

Fokko reviewed (approved):
Some small nit, but apart from that, this looks great 🙌  Thanks @jbonofre

bryanck commented:
> Big picture question I have though is if it makes sense for Iceberg to even be releasing a separate Kafka Connect distribution with HMS client. Since we're close to a release, and I also just don't have context on the rationale I think the change to have separate license/notice for each distribution makes sense cc @bryanck .

The original reason for having a distribution without the HMS client was that Confluent has strict requirements for connectors hosted on Confluent Hub, and Hive with its transitive dependencies has too many security vulnerabilities to fix. In order to have the connector on Confluent Hub we had to strip out the HMS client, but there are several users who are using the HMS support.

We could take the Flink approach, and leave it up to the user to provide the Hive and Hadoop libraries on the classpath. It makes it more difficult and cumbersome to use the connector, however, especially on a hosted platform like AWS MSK or Confluent Cloud.

amogh-jahagirdar commented:
> > Big picture question I have though is if it makes sense for Iceberg to even be releasing a separate Kafka Connect distribution with HMS client. Since we're close to a release, and I also just don't have context on the rationale I think the change to have separate license/notice for each distribution makes sense cc @bryanck .
> 
> The original reason for having a distribution without the HMS client was that Confluent has strict requirements for connectors hosted on Confluent Hub, and Hive with its transitive dependencies has too many security vulnerabilities to fix. In order to have the connector on Confluent Hub we had to strip out the HMS client, but there are several users who are using the HMS support.
> 
> We could take the Flink approach, and leave it up to the user to provide the Hive and Hadoop libraries on the classpath. It makes it more difficult and cumbersome to use the connector, however, especially on a hosted platform like AWS MSK or Confluent Cloud.
> 
> Alternatively, we could have one distribution with HMS client. We could possibly work with Confluent to resolve the security concerns if we ever want to host it on Confluent Hub.

Got it, thanks for the context! Yeah I can imagine for users using HMS client, without Iceberg providing a bundle, it just becomes difficult to setup/use the connector.

amogh-jahagirdar commented:
Thank you @jbonofre, I'll go ahead and merge, thanks @Fokko @bryanck for reviewing and providing context!
